### PR TITLE
Completed work with CRUD operation for alliances

### PR DESCRIPTION
**Pull request by RomaMocherniuk**

### CODE REVIEW

**Review by starovoit777 — APPROVED** [review]

**Comment by maximenkom** [comment]
substitute sysouts with loggers

File: src/main/java/ua/cv/tim/configuration/WebSecurityConfiguration.java [hunk 1]
```diff
@@ -33,22 +42,50 @@ protected void configureGlobalSecurity(AuthenticationManagerBuilder auth) throws
 	protected void configure(HttpSecurity http) throws Exception {
 		http.authorizeRequests()
 				.antMatchers("/logout").authenticated()
-				.antMatchers("/admin**").access("hasRole('ADMIN')")
-				.antMatchers("/leader**").access("hasRole('LEADER')")
-				.antMatchers("/user**").access("hasRole('LEADER') or hasRole('USER')")
+				.antMatchers("/admin/**").access("hasRole('ADMIN')")
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/configuration/WebSecurityConfiguration.java]: add ** after admin and leader -> /admin**/**

File: src/main/java/ua/cv/tim/dao/hibernate/AllianceDaoImpl.java [hunk 2]
```diff
@@ -28,12 +28,38 @@ public Alliance getById(String id) {
...
+        Query query = null;
+        if (uuid != null) {
+            query = session.createQuery("select a from Alliance a where a.name = :name and a.uuid != :uuid");
+            query.setParameter("name", name);
+            query.setParameter("uuid", uuid);
+        } else {
+            query = session.createQuery("select a from Alliance a where a.name = :name");
+            query.setParameter("name", name);
+        }
+        Alliance alliance = (Alliance) query.uniqueResult();
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/dao/hibernate/AllianceDaoImpl.java]: unidueResult() method deprecated. Use getSingleResult() method instead

File: src/main/java/ua/cv/tim/dao/hibernate/UserDaoImpl.java [hunk 3]
```diff
@@ -42,11 +42,23 @@ public User getWithRolesById(String id) {
...
+        if (uuid != null) {
+            System.out.println("uuid no = null : " + uuid);
+            query = session.createQuery("select u FROM User u WHERE u.email=:mail and u.uuid != :uuid");
+            query.setParameter("mail", mail);
+            query.setParameter("uuid", uuid);
+        } else {
+            query = session.createQuery("select u FROM User u WHERE u.email=:mail");
+            query.setParameter("mail", mail);
+        }
+        User user = (User) query.uniqueResult();
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/dao/hibernate/UserDaoImpl.java:60]: above-menthioned

File: src/main/java/ua/cv/tim/dto/AllianceDTO.java [hunk 4]
```diff
@@ -57,11 +66,30 @@ public void setLeaderEmail(String leaderEmail) {
...
+
+    public void setLeaderUuid(String leaderUuid) {
+        this.leaderUuid = leaderUuid;
     }
 
-    public void setUuid(String uuid) {
-        this.uuid = uuid;
+    @Override
+    public String toString() {
+        return "AllianceDTO{" +
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/dto/AllianceDTO.java]: I think better using StringBuilder

Comment (RomaMocherniuk) [reply src/main/java/ua/cv/tim/dto/AllianceDTO.java]: Ok, I will fix it.

**Pull request merged**